FESS-GUI/flywheeloperation.cpp

```diff
@@ -1,58 +1,258 @@
+// Custom Libraries
 #include "flywheeloperation.h"
-#include <cmath>
-#include <QTime>
-#include <QtMath>
 
-FlywheelOperation::FlywheelOperation()
+FlywheelOperation::FlywheelOperation(Interface *inter)
 {
     upperDisplacement = new QPointF();
     lowerDisplacement = new QPointF();
     rotationalPosition = new QPointF();
+
+    interface = inter;
+    setDefaults();
 }
 
-void FlywheelOperation::setMotion(float velocity = NAN, float acceleration = NAN, float jerk = NAN)
+void FlywheelOperation::setMotion(float velocity, float acceleration, float jerk)
 {
+    interface->pushCommand(COMMAND_SET_ALLD_FLOA);
+    interface->pushData(velocity);
+    interface->pushData(acceleration);
+    interface->pushData(jerk);
+}
 
+void FlywheelOperation::setVelocity(float velocity)
+{
+    interface->pushCommand(COMMAND_SET_VELO_FLOA);
+    interface->pushData(velocity);
+}
+
+void FlywheelOperation::setAcceleration(float acceleration)
+{
+    interface->pushCommand(COMMAND_SET_ACCE_FLOA);
+    interface->pushData(acceleration);
+}
+
+void FlywheelOperation::setJerk(float jerk)
+{
+    interface->pushCommand(COMMAND_SET_JERK_FLOA);
+    interface->pushData(jerk);
 }
 
 float FlywheelOperation::getVelocity()
 {
-    double key = QDateTime::currentDateTime().toMSecsSinceEpoch()/1000.0;
-    return qSin(key*1.6+qCos(key*1.7)*2)*10 + qSin(key*1.2+0.56)*20 + 26;
+    float val = vel.front();
```

Comment:
Values get inserted into queues to allow for async access to the flywheel controller's different sensors. 
Example: Velocity is sampled every 90ms and Displacement is sampled every 50ms. If we read every 50ms we would get the current values for Displacement but we wouldn't have values for Velocity. By using a queue we can keep redundant copies of velocities to make up for the difference in timing.  

As long as the GUI sample rate exceeds the fastest sampling rate in the controller the values returned will always be the most recent.